config/src/config.rs

```diff
@@ -273,12 +274,23 @@ impl Clone for BaseConfig {
 }
 
 #[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(default)]
 pub struct MetricsConfig {
     pub dir: PathBuf,
     pub collection_interval_ms: u64,
     pub push_server_addr: String,
 }
 
+impl Default for MetricsConfig {
+    fn default() -> MetricsConfig {
+        MetricsConfig {
+            dir: PathBuf::from("metrics"),
```

Comment:
Do you want to aggregate all the default values somewhere as constants?

Reply:
You mean something like this, with everything specified together?

`const DEFAULT_METRICS_DIR: &str = "metrics";`

I was thinking that having the defaults near each struct definition would make sense (e.g. adding a new setting wouldn't require jumping around), but I don't feel strongly about this.

Reply:
I also don't feel strong.